miniapp/miniapp_util.h

```diff
@@ -28,31 +32,76 @@ enum class ExecutionStrategy { CPU, CUDA, OPENMP };
 */
 namespace memoryManager
 {
-   template <typename T>
-   T *allocate(RAJA::Index_type size)
-   {
-      T *ptr;
+  template <typename T>
+  T *allocate(RAJA::Index_type size, bool host)
+{
+  T *ptr;
 #if defined(RAJA_ENABLE_CUDA)
-      cudaErrchk(
-         cudaMallocManaged((void * *) &ptr, sizeof(T) * size, cudaMemAttachGlobal));
-#else
-      ptr = new T[size];
+  if (!host) {
+     cudaErrchk(
+       cudaMallocManaged((void **)&ptr, sizeof(T) * size, cudaMemAttachGlobal));
+  }
+  else 
+#elif defined(RAJA_ENABLE_HIP)
+  if (!host) {
+     hipErrchk(hipMalloc((void **)&ptr, sizeof(T) * size));
+  }
+  else
 #endif
-      return ptr;
-   }
-
-   template <typename T>
-   void deallocate(T *&ptr)
-   {
-      if (ptr) {
+  {
+    ptr = new T[size];
+  }
+  return ptr;
+}
+
+template <typename T>
+void deallocate(T *&ptr, bool host)
+{
+   if (ptr){
 #if defined(RAJA_ENABLE_CUDA)
+      if (!host) {
          cudaErrchk(cudaFree(ptr));
-#else
-         delete[] ptr;
+      }
+      else
+#elif defined(RAJA_ENABLE_HIP)
+      if (!host) {
+         hipErrchk(hipFree(ptr));
+      }
+      else
 #endif
-         ptr = nullptr;
+      {
+         delete[] ptr;
       }
+      ptr = nullptr;
    }
+}
+
+#if defined(RAJA_ENABLE_CUDA) || defined(RAJA_ENABLE_HIP)
+  template <typename T>
+  T *allocate_gpu(RAJA::Index_type size)
+  {
+    T *ptr;
+#if defined(RAJA_ENABLE_CUDA)
+    cudaErrchk(cudaMalloc((void **)&ptr, sizeof(T) * size));
+#elif defined(RAJA_ENABLE_HIP)
+    hipErrchk(hipMalloc((void **)&ptr, sizeof(T) * size));
+#endif
+    return ptr;
+  }
+
+  template <typename T>
+  void deallocate_gpu(T *&ptr)
+  {
+    if (ptr) {
+#if defined(RAJA_ENABLE_CUDA)
+      cudaErrchk(cudaFree(ptr));
+#elif defined(RAJA_ENABLE_HIP)
+      hipErrchk(hipFree(ptr));
+#endif
+      ptr = nullptr;
+    }
+  }
+#endif
 }; // namespace memoryManager
 
 // Copyright (c) 2016-19, Lawrence Livermore National Security, LLC.
```

Comment:
Is all of this copyright boilerplate up to date and what you still want to have in there? Should the copyright dates get updated?

Reply:
The boilerplate for the below should still be fine. I might need to update the copyright to 2022 though as that's when I updated things to a more modern version of what the raja team had been using...